pkg/app/piped/cmd/piped/piped.go

```diff
@@ -386,8 +386,23 @@ func (p *piped) run(ctx context.Context, t cli.Telemetry) (runErr error) {
 
 	// Start running planpreview handler.
 	if p.enablePlanPreview {
+		// Initialize a dedicated git client for plan-preview feature.
+		// Basically, this feature is an utility so it should not share any resource with the main components of piped.
+		gc, err := git.NewClient(cfg.Git.Username, cfg.Git.Email, t.Logger)
+		if err != nil {
+			t.Logger.Error("failed to initialize git client for plan-preview", zap.Error(err))
+			return err
+		}
+		defer func() {
+			if err := gc.Clean(); err != nil {
+				t.Logger.Error("had an error while cleaning gitClient for plan-preview", zap.Error(err))
+			} else {
+				t.Logger.Info("successfully cleaned gitClient for plan-preview")
+			}
```

Comment:
nit: but it's on you
```suggestion
			if err := gc.Clean(); err != nil {
				t.Logger.Error("had an error while cleaning gitClient for plan-preview", zap.Error(err))
				return
			}
			t.Logger.Info("successfully cleaned gitClient for plan-preview")
```

Reply:
Sure. Applied it.